File: trunk/csharp/UrakawaToolkit/property/channel/TextChannel.cs
using System;
using System.Collections.Generic;
using System.Text;
using urakawa.media;

namespace urakawa.property.channel
{
	/// <summary>
	/// <see cref="Channel"/> that
	/// </summary>
	public class TextChannel : Channel
	{
		internal TextChannel(ChannelsManager chMgr) : base(chMgr) { }

		/// <summary>
		/// Determines if a given <see cref="IMedia"/> can be accepted by the channel,
		/// which it can if it implements interface <see cref="ITextMedia"/>
		/// </summary>
		/// <param name="m">The given media</param>
		/// <returns>A <see cref="bool"/> indicating if the given media can be accepted</returns>
		public override bool canAccept(urakawa.media.IMedia m)
		{
			if (!base.canAccept(m)) return false;
			if (m is ITextMedia) return true;
			return false;
		}
	}
}

File: java/core/src/org/daisy/urakawa/property/channel/TextChannel.java
package org.daisy.urakawa.property.channel;

import org.daisy.urakawa.exception.MethodParameterIsNullException;
import org.daisy.urakawa.media.Media;
import org.daisy.urakawa.media.SequenceMedia;
import org.daisy.urakawa.media.TextMedia;

/**
 *
 */
public class TextChannel extends ChannelImpl {
	/**
	 * @param chMgr
	 * @throws MethodParameterIsNullException
	 */
	public TextChannel(ChannelsManager chMgr)
			throws MethodParameterIsNullException {
		super(chMgr);
	}

	@Override
	public boolean canAccept(Media m) throws MethodParameterIsNullException {
		if (!super.canAccept(m))
			return false;
		if (m instanceof TextMedia)
			return true;
		if (m instanceof SequenceMedia) {
			for (Media sm : ((SequenceMedia) m).getListOfItems()) {
				if (!(sm instanceof TextMedia))
					return false;
			}
			return true;
		}
		return false;
	}
}
